rts/Lua/LuaSyncedCtrl.cpp

```diff
@@ -1754,29 +1756,99 @@ int LuaSyncedCtrl::TransferUnit(lua_State* L)
 	if (unit == nullptr)
 		return 0;
 
-	const int newTeam = luaL_checkint(L, 2);
-	if (!teamHandler.IsValidTeam(newTeam))
+	const int newTeamId = luaL_checkint(L, 2);
+	if (!teamHandler.IsValidTeam(newTeamId))
 		return 0;
 
-	const CTeam* team = teamHandler.Team(newTeam);
-	if (team == nullptr)
+	CTeam* newTeam = teamHandler.Team(newTeamId);
+	if (newTeam == nullptr)
 		return 0;
 
 	bool given = true;
 	if (FullCtrl(L) && lua_isboolean(L, 3))
 		given = lua_toboolean(L, 3);
 
+	bool adjustUnitLimit = false;
```

Comment:
You can simplify these 3 lines to:

```cpp
bool adjustUnitLimit = FullCtrl(L) && luaL_optboolean(L, 4, false)
```

Reply:
Same for the check right above if you don't mind, having the explicit default in optboolean improves readability although that can be argued I guess.

Reply:
I don't think FullCtrl is needed at all.

Reply:
Because Synced environment?